### PR TITLE
[Close?] LF-3618: Create swipeable summary icon tiles component

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- update `Tile` component to take `info` prop
- update `Tiles` component
   - take `maxSwipeableWidth`
   - listen to the window size change and add a className that achieves swipeable view (this [article](https://css-tricks.com/working-with-javascript-media-queries/) was helpful)
- add stories (http://localhost:6006/?path=/story/components-tile-tiles--swipeable-label-icon-info-tiles)
   (Please adjust `maxSwipeableWidth` or window size to make it swipeable) 
  Screenshot 2023-09-14 at 4 39 34 PM
- create mixin to truncateText
- fix className `taskTypeLabelContainer` that was in `IconLabelTile`

Note: You will notice that the info text is cut, but I will leave it for later to adjust it since we don't have a mockup for desktop view.
Screenshot 2023-09-14 at 4 45 33 PM

Jira link: https://lite-farm.atlassian.net/browse/LF-3618

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/46a99d1a-8fb5-4596-a526-fdd7f39c3af7

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Tile/Tiles.jsx [hunk 1]
```diff
@@ -25,9 +26,44 @@ const tileComponents = {
...
+    }
+
+    // https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia
+    const media = window.matchMedia(queryToMatch);
+
+    if (media.matches !== isSwipableView) {
+      setIsSwipableView(media.matches);
+    }
+
+    const listner = () => {
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Tile/Tiles.jsx:57]: small typo listener

**Comment by Duncan-Brain** [comment]
It looks really good! And a lot simpler than I thought it would be --  I am glad you know what to do ;) . 

The two things I noticed I think you covered with the description: I was expecting it to be swipeable always on desktop -- but I think you are waiting for a different desktop view:
Screenshot 2023-09-19 at 10 24 46 AM

**Comment by SayakaOno** [comment]
@Duncan-Brain Thank you Duncan! I felt like I saw the desktop view that had the summary on the left and tiles on the right... it doesn't matter anymore though 🤣

**Comment by kathyavini** [comment]
😢 
 Screenshot 2023-09-20 at 7 59 45 AM

**Pull request closed without merging**